TeapotTests/DeliveryQueueTests.swift

```diff
@@ -0,0 +1,189 @@
+//
+//  DeliveryQueueTests.swift
+//  TeapotMac
+//
+//  Created by Igor Ranieri on 21.09.18.
+//  Copyright © 2018 B&B. All rights reserved.
+//
+
+@testable import TeapotMac
+import XCTest
+
+class DeliveryQueueTests: XCTestCase {
+
+    func testDeliveryOnQueueGet() {
+        let expectation = self.expectation(description: "Get")
+
+        let dispatchQueue = DispatchQueue(label: "dispatch.queue.here")
+        let teapot = Teapot(baseURL: URL(string: "https://httpbin.org")!,
+                            deliveryQueue: dispatchQueue)
+
+        teapot.get("/get") { result in
+            XCTAssertTrue(!Thread.isMainThread)
+            let queueLabel = String(cString: __dispatch_queue_get_label(nil), encoding: .utf8)!
```

Comment:
oh this seems super legit 😄  - i assume this is undocumented?

Reply:
Well, I forgot that tests run agains a Mac target, where we have a lot more freedom. Yes, this is documented... ish... https://developer.apple.com/documentation/dispatch/1452939-dispatch_queue_get_label

Reply:
Ah nice.